fuel-txpool/src/txpool.rs

```diff
@@ -205,6 +207,12 @@ impl TxPool {
             Transaction::Create(create) => create.price(),
             Transaction::Mint(_) => unreachable!(),
         };
+        if self.config.metrics {
+            // Gas Price metrics are recorded here to avoid double matching for
+            // every single transaction, but also means metrics aren't collected on gas
+            // price if there is no minimum gas price
+            TXPOOL_METRICS.gas_price_histogram.observe(price as f64);
```

Comment:
Do we want to record data for potentially invalid transactions? Maybe better do it only for committed transactions(during committing the block)? Or have two metrics for invalid and valid transactions=)

Reply:
There reason I did it here was to avoid checking price twice on each transaction, I can move it down to just before it is actually inserted into the txpool. I do think it might be worth exploring metrics for invalid tx's though for insights like if the pool is being spammed with a bunch of low gas txs so I might tackle that in a subsequent PR

Reply:
moved this